torch_xla/experimental/_distributed_checkpoint_helpers.py

```diff
@@ -186,4 +190,22 @@ def narrow_tensor_by_index(tensor: torch.Tensor, offsets: Sequence[int],
       # recording here for the narrow op and 'local_shard' should be a
       # leaf variable in the autograd graph.
       narrowed_tensor = narrowed_tensor.narrow(idx, offset, size)
-  return narrowed_tensor
\ No newline at end of file
+  return narrowed_tensor
+
+
+@dataclasses.dataclass
+class _CpuShards:
+  shards: List[xs.XLAShard]
+  global_shape: torch.Shape
+
+
+def _sharded_cpu_state_dict(state_dict: STATE_DICT_TYPE) -> STATE_DICT_TYPE:
+  """
+  Converts a state_dict on XLA device to a sharded state_dict on CPU.
+  """
+
+  def move_state_dict_to_cpu(v):
+    v = xs.wrap_if_sharded(v)
+    return _CpuShards(shards=v.local_shards, global_shape=v.global_tensor.shape)
```

Comment:
We should only wrap tensors wit non-REPLICATED sharding, similar to https://github.com/pytorch/xla/blob/42d92709b58ad86a12ba94b834399ead1a6392f4/torch_xla/experimental/distributed_checkpoint.py#L378-L381. You can move that method into this file to avoid circular dependencies.